packages/flutter/test/widgets/mouse_region_test.dart

```diff
@@ -1420,6 +1444,206 @@ void main() {
     expect(logs, <String>['paint', 'hover-enter']);
   });
 
+  testWidgets('Changing MouseRegion.cursor is effective and repaints', (WidgetTester tester) async {
+    final List<String> logPaints = <String>[];
+    final List<String> logEnters = <String>[];
+
+    final TestGesture gesture = await tester.createGesture(kind: PointerDeviceKind.mouse);
+    await gesture.addPointer(location: const Offset(100, 100));
+    addTearDown(gesture.removePointer);
+
+    final VoidCallback onPaintChild = () { logPaints.add('paint'); };
+
+    await tester.pumpWidget(_Scaffold(
+      topLeft: Container(
+        height: 10,
+        width: 10,
+        child: MouseRegion(
+          cursor: SystemMouseCursors.forbidden,
+          onEnter: (_) { logEnters.add('enter'); },
+          opaque: true,
+          child: CustomPaint(painter: _DelegatedPainter(onPaint: onPaintChild)),
+        ),
+      ),
+    ));
+    await gesture.moveTo(const Offset(5, 5));
+
+    expect(logPaints, <String>['paint']);
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.forbidden);
+    expect(logEnters, <String>['enter']);
+    logPaints.clear();
+    logEnters.clear();
+
+    await tester.pumpWidget(_Scaffold(
+      topLeft: Container(
+        height: 10,
+        width: 10,
+        child: MouseRegion(
+          cursor: SystemMouseCursors.text,
+          onEnter: (_) { logEnters.add('enter'); },
+          opaque: true,
+          child: CustomPaint(painter: _DelegatedPainter(onPaint: onPaintChild)),
+        ),
+      ),
+    ));
+
+    expect(logPaints, <String>['paint']);
+    expect(RendererBinding.instance.mouseTracker.debugDeviceActiveCursor(1), SystemMouseCursors.text);
+    expect(logEnters, isEmpty);
+    logPaints.clear();
+    logEnters.clear();
+  });
+
+  testWidgets('Changing whether MouseRegion.cursor is null is effective and repaints', (WidgetTester tester) async {
```

Comment:
Do you have a test to verify that the cursor of the MouseRegion underneath the one that changed it cursor to null becomes active?

Reply:
Good idea. I changed this test by adding a background `MouseRegion`